File: two-pointer/valid-palindrome/validPalindrome.js
function isPalindrome(s) {
  let left = 0;
  let right = s.length - 1;

  while (left < right) {
    while (left < right && !s[left].isalnum()) {
      left++;
    }
    while (left < right && !s[right].isalnum()) {
      right--;
    }

    if (s[left].toLowerCase() !== s[right].toLowerCase()) {
      return false;
    }
  }
}

File: two-pointer/valid-palindrome/valid_palindrome.py
def isPalindrome(s: str):
  left, right = 0, len(s) - 1

  while left < right:
    while left < right and not s[left].isalnum():
      left += 1
    while left < right and not s[right].isalnum():
      right -= 1

    if s[left].lower() != s[right].lower():
      return False
  pass


s = "A man, a plan, a canal: Panama"
print(isPalindrome(s))
